src/main/scala/edu/berkeley/ce/rockslicing/SeedJointSelector.scala

```diff
@@ -0,0 +1,251 @@
+package edu.berkeley.ce.rockslicing
+
+import scala.annotation.tailrec
+
+/**
+  * Selects joints from the input joint set that will be used to seed the RDD in order to maintain
+  * approximately balanced loads among processors.
+  */
+object SeedJointSelector {
+  val THRESHOLD = 0.50
+
+  /**
+    * Searches though sequence of joint sets to find best seed joints
+    *
+    * @param jointSets Seq of Seq's of Joints representing input joint sets
+    * @param inputVolume Block representing input rock volume
+    * @param numProcessors Number of processors to be used in the analysis
+    * @param remainingJoints Seq of Seq of joints that were not selected as seed joints
+    * @return Returns a tuple containing two Seq's of Joints. The first Seq contains the seed joints that
+    *         best divide the input volume. The second Seq contains the joints that remain from the input
+    *         joint set.
+    */
+  @tailrec
+  def searchJointSets(jointSets: Seq[Seq[Joint]], inputVolume: Block, numProcessors: Int,
+                      remainingJoints: Seq[Seq[Joint]] = Seq.empty[Seq[Joint]]): (Seq[Joint], Seq[Joint]) = {
+    val allJoints = if (remainingJoints.isEmpty) {
```

Comment:
This is essentially what the code was trying to do. I think where it got confusing is that I was just keeping the entire input joint set saved in `allJoints` -  this way `remainingJoints` was really just the entire input joint set after the first function call. Yes, this is confusing! The way you suggest returning after finding a suitable joint set is what the function was doing, but I think because of the confusion about `remainingJoints` it wasn't clear.
I've updated the function so that the `if` at the beginning is gone and that `remainingJoints` is actually just the remaining joints. `remainingJoints` and `allOtherJoints` - which is what is returned - maintain the ordering of the joint sets in the input joint set.